vignettes/msnset.Rmd

```diff
@@ -0,0 +1,470 @@
+---
+title: "Working with MSnSets"
+author: "Charlotte Dawson"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+bibliography: camprot_references.json
+vignette: >
+  %\VignetteIndexEntry{Working with MSnSets}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r, message=FALSE}
+library(MSnbase)
+library(camprotR)
+library(dplyr)
+```
+
+## Introduction
+
+**What is an MSnSet?** To quote from `MSnbase`:
+
+> The `MSnSet` class is derived form the `Biobase::eSet` class and mimics the
+`Biobase::ExpressionSet` class classically used for microarray data.
+
+This description is a bit dense and unintelligible to the uninitiated.
+Here, I will describe an MSnSet in my own words.
+
+
+MSnSets contain information about an MS experiment. To better understand MSnSets
+we first need to define some terminology.
+
+In a quantitative proteomics experiment we are analysing 'samples' from
+different experimental conditions via MS, e.g. comparing 'samples' from cells treated 
+with a drug versus a control. The quantitative data we eventually
+obtain consists of measurements of 'features'. In this context, features can be 
+PSMs, peptides, or proteins.
+
+- An MSnSet is a special type of list (specifically it is an S4 object) that 
+describes an MS experiment. 
+- This list contains multiple objects of different types:
+  - The quantitative data ('assay data'; in a `matrix`)
+  - The associated 'feature data' for each row in the quantification data (in a `Biobase::AnnotatedDataFrame`)
+  - Metadata describing the MS experiment ('phenotype Data'; in a `Biobase::AnnotatedDataFrame`)
+- These underlying objects must have a specific structure. For example the sample names in the assay data (columns) must match the sample names in the phenotype data (rownames) and the assay data and feature data must have the same number of rows.
+
+Conveniently, the `MSnbase` package comes with some example MSnSets. In this
+section we will explore the `msnset` MSnSet. This data set is from an 
+iTRAQ 4-plex experiment wherein BSA and Enolase have been spiked into a background
+of _Erwinia_ proteins. See `?msnset` for more information.
+
+## Exploring an MSnSet
+
+Below we make an example list to compare to an MSnSet.
```

Comment:
I'm unclear what the benefit of defining this list, just to explore the structure is. Personally, I'd drop this part and just go straight to `str(msnset)`